FirebaseAppDistribution/Sources/FIRAppDistributionAuthPersistence.m

```diff
@@ -16,77 +16,99 @@
-    }
-    return errSecSuccess || status == errSecItemNotFound;
++ (BOOL)clearAuthState:(NSError **_Nullable)error {
+  NSMutableDictionary *keychainQuery = [self getKeyChainQuery];
+  OSStatus status = SecItemDelete((CFDictionaryRef)keychainQuery);
+
+  if (status != errSecSuccess && status != errSecItemNotFound && error) {
+    NSString *desc = NSLocalizedString(
+        @"Failed to clear auth state from keychain. Tester will overwrite data on sign in.",
+        @"Error message for failure to retrieve auth state from keychain");
+    NSDictionary *userInfo = @{NSLocalizedDescriptionKey : desc};
+    *error = [NSError errorWithDomain:kFIRAppDistributionKeychainErrorDomain
+                                 code:FIRAppDistributionErrorTokenDeletionFailure
+                             userInfo:userInfo];
+  }
+  return errSecSuccess || status == errSecItemNotFound;
 }
 
-+ (OIDAuthState*)retrieveAuthState {
-    NSMutableDictionary *keychainQuery = [self getKeyChainQuery];
-      [keychainQuery setObject:(id)kCFBooleanTrue forKey:(id)kSecReturnData];
-      [keychainQuery setObject:(id)kSecMatchLimitOne forKey:(id)kSecMatchLimit];
-      CFDataRef passwordData = NULL;
-      NSData *result = nil;
-      OSStatus status = SecItemCopyMatching((CFDictionaryRef)keychainQuery,
-                                         (CFTypeRef *)&passwordData);
-      if (status == noErr && 0 < [(__bridge NSData *)passwordData length]) {
-        result = [(__bridge NSData *)passwordData copy];
-      } else {
-          NSLog(@"AUTH ERROR - cannot lookup keystore config");
-      }
-      if (passwordData != NULL) {
-        CFRelease(passwordData);
-      }
-    
-    OIDAuthState *authState = nil;
-    if(result) {
-        authState = (OIDAuthState *)[NSKeyedUnarchiver unarchiveObjectWithData:result];
-    }
-    
-    return authState;
++ (OIDAuthState *)retrieveAuthState:(NSError **_Nullable)error {
+  NSMutableDictionary *keychainQuery = [self getKeyChainQuery];
+  [keychainQuery setObject:(id)kCFBooleanTrue forKey:(id)kSecReturnData];
+  [keychainQuery setObject:(id)kSecMatchLimitOne forKey:(id)kSecMatchLimit];
+  CFDataRef passwordData = NULL;
+  NSData *result = nil;
+  OSStatus status = SecItemCopyMatching((CFDictionaryRef)keychainQuery, (CFTypeRef *)&passwordData);
+
+  if (status == noErr && 0 < [(__bridge NSData *)passwordData length]) {
+    result = [(__bridge NSData *)passwordData copy];
+  } else if (error) {
+    NSString *desc = NSLocalizedString(
+        @"Failed to retrieve auth state from keychain. Tester will have to sign in again.",
+        @"Error message for failure to retrieve auth state from keychain");
+    NSDictionary *userInfo = @{NSLocalizedDescriptionKey : desc};
+    *error = [NSError errorWithDomain:kFIRAppDistributionKeychainErrorDomain
+                                 code:FIRAppDistributionErrorTokenRetrievalFailure
+                             userInfo:userInfo];
+  }
```

Comment:
2 things:
 - error is a pointer to a pointer. This should probably be checking that the pointer we're setting (*error) is not null. Or else this will always be true.
 - This seems odd to re-set the error when you pass an error to this method. Could it be we forgot to rename a variable that got deleted?

Reply:
Nevermind - you want to see if we passed in an error variable... Ignore this